org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/fix/JoinVariableFixCore.java

```diff
@@ -0,0 +1,315 @@
+			return null;
+		}
+
+		if (!(fragment.getParent() instanceof VariableDeclarationStatement)) {
+			return null;
+		}
+		VariableDeclarationStatement statement= (VariableDeclarationStatement) fragment.getParent();
+
+		SimpleName[] names= LinkedNodeFinder.findByBinding(statement.getParent(), binding);
+		if (names.length <= 1 || names[0] != fragment.getName()) {
+			return null;
+		}
+		SimpleName firstAccess= names[1];
+		if (onFirstAccess) {
+			if (firstAccess != node) {
+				return null;
+			}
+		} else {
+			if (firstAccess.getLocationInParent() != Assignment.LEFT_HAND_SIDE_PROPERTY) {
+				return null;
+			}
+		}
+		Assignment assignment= (Assignment) firstAccess.getParent();
+		if (assignment.getLocationInParent() != ExpressionStatement.EXPRESSION_PROPERTY) {
+			return null;
+		}
+		ExpressionStatement assignParent= (ExpressionStatement) assignment.getParent();
+		IfStatement ifStatement= null;
+		Expression thenExpression= null;
+		Expression elseExpression= null;
+		ITypeBinding exprBinding= null;
+
+		ASTNode assignParentParent= assignParent.getParent();
+		if (assignParentParent instanceof IfStatement
+				|| (assignParentParent.getLocationInParent() == IfStatement.THEN_STATEMENT_PROPERTY
+						&& !(assignParentParent.subtreeMatch(new ASTMatcher(), statement.getParent())))) {
+			if (assignParentParent.getLocationInParent() == IfStatement.THEN_STATEMENT_PROPERTY) {
+				assignParentParent= assignParentParent.getParent();
+			}
+			ifStatement= (IfStatement) assignParentParent;
+			Statement thenStatement= getSingleStatement(ifStatement.getThenStatement());
+			Statement elseStatement= getSingleStatement(ifStatement.getElseStatement());
+			if (thenStatement == null || elseStatement == null) {
+				return null;
+			}
+
+			if (thenStatement instanceof ExpressionStatement && elseStatement instanceof ExpressionStatement) {
+				Expression inner1= ((ExpressionStatement) thenStatement).getExpression();
+				Expression inner2= ((ExpressionStatement) elseStatement).getExpression();
+				if (inner1 instanceof Assignment && inner2 instanceof Assignment) {
+					Assignment assign1= (Assignment) inner1;
+					Assignment assign2= (Assignment) inner2;
+					Expression left1= assign1.getLeftHandSide();
+					Expression left2= assign2.getLeftHandSide();
+					if (left1 instanceof Name && left2 instanceof Name && assign1.getOperator() == assign2.getOperator()) {
+						IBinding bind1= ((Name) left1).resolveBinding();
+						IBinding bind2= ((Name) left2).resolveBinding();
+						if (bind1 == bind2 && bind1 instanceof IVariableBinding) {
+							exprBinding= ((IVariableBinding) bind1).getType();
+							thenExpression= assign1.getRightHandSide();
+							elseExpression= assign2.getRightHandSide();
+						}
+					}
+				}
+			}
+			if (thenExpression == null || elseExpression == null) {
+				return null;
+			}
+		} else {
+			// Be conservative and don't allow anything but Blocks between the
```

Comment:
@jjohnstn I'm not really sure about this change I did instead of just copying what was there. I changed it because the original didn't make sense to be with the loop. With this change I see that

```
List list;

{
    list = new ArrayList();
}
```

get joined properly. It works even there are nested blocks. Hope this is what the original code's intension was too.

Reply:
Looks fine